Services/BaseService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using PSP.Models;
+using PSP.Repositories;
+
+namespace PSP.Services
+{
+    public abstract class BaseService<T, TCreate> : IBaseService<T, TCreate> where T : class where TCreate : class
```

Comment:
more just thinking out loud here,
very good idea to have these crud operations in a base class to reduce code duplication, but I'm not sure if base service is the right place - services can hold logic not only for crud operations, but also various business logic. So maybe rename this to something else. if we'll have service for calculating price of order with discounts or whatever, it won't make sense to have `GetAll()`, `Get()`, etc. but you would expect that class to inherit `BaseService`

Reply:
Good point, renamed to `CrudEntityService`. Let me know if you think this change is enough.